cmake/kokkos_tribits.cmake

```diff
@@ -136,13 +136,22 @@ ENDFUNCTION()
 
 FUNCTION(KOKKOS_ADD_EXECUTABLE_AND_TEST ROOT_NAME)
 IF (KOKKOS_HAS_TRILINOS)
+  CMAKE_PARSE_ARGUMENTS(PARSE
+    ""
+    ""
+    "SOURCES;CATEGORIES;ARGS"
+    ${ARGN})
+  IF(DEFINED PARSE_ARGS)
+    STRING(REPLACE ";" " " PARSE_ARGS "${PARSE_ARGS}")
```

Comment:
I don't think this is necessary here? You should be able to pass this through directly as a list. Were you getting some sort of error?

Reply:
This was necessary to make sure all three command-line arguments for the `UnitTest_CMakePassCmdLineArgs` test were properly passed to it by TriBITS; without this, 3 separate unit tests were created, one per argument, with each test failing (the unit test verifies all three arguments as the passing condition).

I was confused by the problem but did a check on the TriBITS documentation and it says this about multiple arguments, I'm guessing this is why this change resolved the issue?

> WARNING: Multiple arguments passed to a single test invocation must be quoted or multiple tests taking single arguments will be created instead! 

https://tribits.org/doc/TribitsDevelopersGuide.html#tribits-add-test

Reply:
Oh, gotcha. I had misunderstood Tribits here.  In that case, we want the function behavior to match TriBITS. We don't want the kokkos wrappers to have different behavior.  We need to change the unit_test/CMakeLists.txt to pass the args as a single string.

Reply:
@jjwilke I tried removing the `STRING(REPLACE ";" " " PARSE_ARGS "${PARSE_ARGS}")` and adding quotes around the args for the UnitTest_CMakePassCmdLineArgs.cpp test in core/unit_test/CMakeLists.txt (i.e. `ARGS "one 2 THREE"`) but running into some whack-a-mole fun; this still passes in Trilinos as expected but fails in a standalone Kokkos build. Any suggestions?

Reply:
Yeah, my bad. I was misunderstanding what TriBITS wanted. I pushed a change that hopefully fixes things.

Reply:
@jjwilke I tested the standalone Kokkos build and I'm seeing configure errors like this:

```
CMake Error at cmake/fake_tribits.cmake:139 (SET_TESTS_PROPERTIES):
  SET_TESTS_PROPERTIES Can not find test to add properties to:
  KokkosCore_UnitTest_Serial1
Call Stack (most recent call first):
  cmake/kokkos_tribits.cmake:171 (KOKKOS_ADD_TEST)
  core/unit_test/CMakeLists.txt:174 (KOKKOS_ADD_EXECUTABLE_AND_TEST)
...#etc
```